src/temporal.ts

```diff
@@ -58,31 +58,18 @@ export const temporalStateCreator = <TState>(
       setOnSave: (_onSave) => set({ _onSave }),
       // Internal properties
       _onSave: options?.onSave,
-      _handleSet: (pastState) => {
+      _handleSet: (pastState, currentState, deltaState) => {
         if (get().isTracking) {
```

Comment:
Should this `if` statement also be moved into `temporalHandleSet()`?

Reply:
Taking a look at this now, I'm wondering if `_handleSet` should simply be removed from the `temporal` store because it's never externally referenced. What do you think?

Reply:
Re removing `_handleSet` -- my apologies, I am not totally confident I understand the potential refactor.

Although `_handleSet` shouldn't be accessed by the user (it is notably omitted from the TemporalState type, along with `_onSave`), `_handleSet` does still need access to `get` and `set` params passed by `createStore` to the `stateCreator` function returned by `temporalStateCreator`. 

And then `_handleSet` is accessed in the `curriedHandleSet` expression via `(store.temporal.getState() as _TemporalState<TState>)._handleSet;`

I'm just trying to puzzle through how `_handleSet` might otherwise have access to `get` and `set` of the temporal store

Reply:
Re moving that `if` statement, yes that makes sense!

Reply:
Yes! Your code snippet is what I had in mind; however, this is simply a hunch that it will work but I haven't tested it. 

We could leave it as a follow-up PR to avoid many changes in a single PR. What do you think about you or I making this change after this current PR is merged?

Reply:
Yeah that sounds good! Happy to make a follow up PR after this one to test it out.